src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java

```diff
@@ -0,0 +1,457 @@
+                                if (!street.getOsmId().equals(wayLabel)) {
+                                    LOG.warn("Mismatched path street edge with way ID {} along way {}", street.getOsmId(), wayLabel);
+                                    goodPath = false;
+                                } else {
+                                    matchedEdges.add(street);
+
+                                    // TODO: surely there's an easier way to find the back edge.
+                                    // add reverse of this edge
+                                    boolean foundReverse = false;
+                                    for (Edge reverse : wayEdges) {
+                                        if (reverse.isReverseOf(street)) {
+                                            foundReverse = true;
+                                            matchedEdges.add((StreetEdge) reverse);
+                                        }
+                                    }
+                                    if (!foundReverse) {
+                                        LOG.warn("No reverse found for matched SPT edge {}: {}", edge.getName(), ((StreetEdge) edge).getOsmId());
+                                    }
+                                }
+                            } else {
+                                LOG.warn("Edge on SPT is not a StreetEdge!  Cannot use it.");
+                            }
+                        }
+                    }
+
+                    options.cleanup();
+
+                    if (!goodPath) {
+                        LOG.warn("Bad path for way {}: {}", wayLabel, wayName);
+                        continue;
+                    }
+                }
+            }
+
+            // sanity check
+            LOG.info("Have {} matched street edges for way {}.", matchedEdges.size(), wayLabel + " " + wayName);
+            boolean isBad = false;
+            for (StreetEdge edge : matchedEdges) {
+                if (!edge.getOsmId().equals(wayLabel)) {
+                    LOG.warn("Found bad matched way!  Got: {} Should be: {}", edge.getOsmId(), wayLabel);
+                    isBad = true;
+                }
+            }
+
+            if (isBad) {
+                continue;
+            }
+
+            ///////////////////////////////////////////////////////////////////////
+            // If got this far, have a good set of matched way edges to work with.
+            // Now go set properties on them.
+            ///////////////////////////////////////////////////////////////////////
+
+            // TODO: Set edge properties
+        }
+
+        LOG.info("Found {} start and {} end intersections.", foundStartCt, foundEndCt);
+        if (missingStartCt > 0 || missingEndCt > 0) {
+            LOG.warn("Missing {} start intersections.", missingStartCt);
+            LOG.warn("Missing {} end intersections.", missingEndCt);
+        }
+    }
```

Comment:
I'm having trouble figuring out what this function affects; does it set the object state, or does it change one of the parameters in place (`graph`, perhaps?). If possible, it would make things clearer to use a temporary variable throughout the function body, and then assign to its ultimate destination at the end so it's easy to find the output. Some additional commenting and/or breaking different phases out into functions would also help make the overall structure easier to read.